enhancements/etcd/dr-leveraging-ceo-scaling.md

```diff
@@ -0,0 +1,188 @@
+
+This document discusses the ideas involved in making the first two scenarios easiest to implement and consumed by the 
+user. The ideas is to have a single member etcd cluster running so control plane is up and CEO scales the remaining
+etcd members.  
+
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Open Questions [optional]
+
+TODO
+
+## Summary
+
+cluster-etcd-operator (CEO) is an operator that handles the scaling of etcd during cluster
+bootstrap and regular operation and provisioning etcd dependencies such as TLS certificates.
+
+## Motivation
+
+TODO
+
+
+### Goals
+
+1. Leverage the scale up mechanism of CEO
+
+### Non-Goals
+
+TODO
+
+## Challenges and possible alternatives
+
+A restore operation is a two step process on each node. 
+1. Restoring the data directory with appropriate environment variables with the following operation:
+```
+        ETCDCTL_API=3 etcdctl snapshot restore ./path/to/backup.db \
+                    --name $ETCD_NAME \
+                    --initial-cluster="$ETCD_NAME=$ETCD_NODE_PEER_URL" \
+                    --initial-cluster-token openshift-etcd \
+                    --initial-advertise-peer-urls $ETCD_NODE_PEER_UR \
+                    --data-dir="/path/to/new/data-dir"
+```
+2. Starting the etcd pod should be started with the same value of $ETCD_NAME, ETCD_INITIAL_CLUSTER and 
+$ETCD_NODE_PEER_URL as in step 1.
+
+This will allow etcd to populate the data directory with a new cluster ID, and new membership data that includes only
+1 member.
+
+Restoring a single member from a backup and scaling up has two distinct problem, they are:
+1. Get the first etcd member running. 
+2. Get the operator to deploy static pods that can decide that they are being restored, so they don't start on the 
+existing data directories.
+
+### Static Pod identifying a restore
+
+Assuming the first member is up, the control plane will be up and operator start running. A user can trigger a new 
+rollout of the etcd pods by running `oc patch forceredpeloymment=reason-time`. New pods will be rolled out on the 
+remaining masters that have to be restored. Before etcd commands on these pods are run, the existing data directory 
+needs to be removed, because they contain the previous cluster state information.
+
+During the start up pod queries etcd membership, if the pod is an unstarted member, it can be assumed that the data 
+directory present is from previous cluster and moved out.
+
+Other solution is to check the cluster id of the new cluster with the cluster id of the existing data directory, move 
+out the data directory only if they are different. This is a little time consuming to implement and test.
```

Comment:
Tools[1] exist that set this foundation to review the cluster bucket of the state file. If we were going to attempt this extra caution needs to be taken in order to ensure we do not run into lock issues with the db and etcd.

[1] https://github.com/etcd-io/etcd/tree/master/tools/etcd-dump-db

Reply:
@hexfusion is the db locked even when etcd is not running? 

From what I have read it is only locked when at least 1 instance of etcd is running so as to prevent the 2nd instance to start on the same data directory. I can dig into the source code to figure out, .